### ISSUE TITLE
Form should accept all form props

### ISSUE THREAD

**Issue by itsMapleLeaf** [smeijer/next-runtime#3]
I need to be able to apply `className` and `action` in some cases; it'd be convenient for the form to accept all the props that a form can accept.

**Comment by smeijer** [comment]
Agreed. 

What's your expected behavior for prop overrides? I mean, what if we would really take `all props` and someone passes the onSubmit handler? Should we invoke our own logic + their handler? Or only theirs?

**Comment by itsMapleLeaf** [comment]
I looked back at the form code, and I see there's already `onSuccess` and `onError`, so I think calling the user's `onSubmit` on submit start would make sense. They might need the event for whatever reason ¯\\\_(ツ)\_/¯

On an unrelated note, I'm enjoying this library a lot so far 😄

**Comment by smeijer** [comment]
I'm working on this, and I've decided to merge the `onSubmit` handlers.

The user-defined onSubmit handler will be called first. If they call `event.preventDefault()`, things come to an end, and `Form` won't handle the submission. If they don't call `event.preventDefault()` the submission will start right after that.

In other words:

```jsx
const handleSubmit = async (event) => {
  onSubmit?.(event);
  if (event.defaultPrevented) return;
  event.preventDefault();

  // submit it
}

<form onSubmit={handleSubmit} {...props} />
```

This makes the `validate` function a bit redundant though (as validation can now be done in `onSubmit`. So I'm not sure if that will stay or has to go.

**Comment by itsMapleLeaf** [comment]
Yeah, I agree that the validate function seems a bit redundant. If the API can be smaller, then make it smaller, I say

**Comment by MichaelDeBoey** [comment]
Removing the `validate` functionality is a breaking change.

@smeijer Also make sure to account for the [`novalidate` attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/form#attr-novalidate) if you would keep the `validate` method

**Comment by itsMapleLeaf** [comment]
I think we can get away with breaking changes a little more easily while the library is fresh, but one could just deprecate it anyways if that's still a concern ¯\\\_(ツ)\_/¯

**Comment by smeijer** [comment]
I've published this as a minor, and it's available under 1.1.0. I don't think it makes sense to publish 2.0 after only 3 days of existence :sweat_smile:

The minor is as per [semantic version](https://semver.org/#what-if-i-inadvertently-alter-the-public-api-in-a-way-that-is-not-compliant-with-the-version-number-change-ie-the-code-incorrectly-introduces-a-major-breaking-change-in-a-patch-release) prescribes.

> Use your best judgment. If you have a huge audience that will be drastically impacted by changing the behavior back to what the public API intended, then it may be best to perform a major version release, even though the fix could strictly be considered a patch release.

This will be the only time that I'll introduce a breaking change like this. But because we are 3 days old and have under 200 installs @ npm, I think it's okay(ish) for now.
